src/profiling/meter_manager.cpp

```diff
@@ -1,34 +1,79 @@
+#include <communication/global_policy.hpp>
+#include <util/hostname.hpp>
+#include <json/json.hpp>
+
 #include "meter_manager.hpp"
+#include "memory_meter.hpp"
+#include "power_meter.hpp"
 
 namespace nest {
 namespace mc {
 namespace util {
 
+measurement::measurement(
+        std::string n, std::string u, const std::vector<double>& readings):
+    name(std::move(n)), units(std::move(u))
+{
+    using gcom = communication::global_policy;
+
+    // Assert that the same number of readings were taken on every domain.
+    const auto num_readings = readings.size();
+    if (gcom::min(num_readings)!=gcom::max(num_readings)) {
+        throw std::out_of_range(
+            "the number of checkpoints in the \""+name+"\" meter do not match across domains");
+    }
+
+    // Gather across all of the domains onto the root domain.
+    for (auto r: readings) {
+        measurements.push_back(gcom::gather(r, 0));
```

Comment:
I think global communication in a constructor is surprising in a bad way, but I will submit a counter-proposal in a subsequent PR.